PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -165,24 +165,17 @@ LanguageNameInfo ScintillaEditView::_langNameInfoArray[L_EXTERNAL + 1] = {
 
 int getNbDigits(int aNum, int base)
 {
-	int nbChiffre = 1;
-	int diviseur = base;
+	int nbDigits = 0;
 
-	for (;;)
+	do
 	{
-		int result = aNum / diviseur;
-		if (!result)
-			break;
-		else
-		{
-			diviseur *= base;
-			++nbChiffre;
-		}
-	}
-	if ((base == 16) && (nbChiffre % 2 != 0))
-		nbChiffre += 1;
+		++nbDigits;
+		aNum /= base;
+	} while (aNum != 0);
+	if (base == 16 && nbDigits % 2 != 0)

```

Comment:
@cddiffz For input aNum = 0 and base =16 this would result in nbDigits = 1 from do-while above already here and as 1%2 = 1 != 0 would result in another increment to nbDigits.

Reply:
But that was already the case before. Ididn't change the semantics at all, just fixed the bug.

Reply:
Yes, that is correct also it seems to be strange.

Reply:
This is another bug for me. With "Format" set to "Hex", e.g. instead of "100", "100 " is inserted because of the rounding to the next even number.

Reply:
See #15168